diff --git a/RWKV-v7/train_temp/src/model.py b/RWKV-v7/train_temp/src/model.py
--- a/RWKV-v7/train_temp/src/model.py
+++ b/RWKV-v7/train_temp/src/model.py
@@ -547,19 +547,19 @@ class RWKV_Tmix_x070(MyModule):
-            D_DECAY_LORA = max(32, int(round(  (2.5*(C**0.5))  /32)*32)) # suggestion
+            D_DECAY_LORA = getattr(args, "decay_lora_rank", max(32, int(round(  (2.5*(C**0.5))  /32)*32))) # suggestion
             self.w1 = nn.Parameter(torch.zeros(C, D_DECAY_LORA))
             self.w2 = nn.Parameter(ortho_init(torch.zeros(D_DECAY_LORA, C), 0.1))
             self.w0 = nn.Parameter(www.reshape(1,1,C) + 0.5 + zigzag*2.5)
 
-            D_AAA_LORA = max(32, int(round(  (2.5*(C**0.5))  /32)*32)) # suggestion
+            D_AAA_LORA = getattr(args, "aaa_lora_rank", max(32, int(round(  (2.5*(C**0.5))  /32)*32))) # suggestion
             self.a1 = nn.Parameter(torch.zeros(C, D_AAA_LORA))
             self.a2 = nn.Parameter(ortho_init(torch.zeros(D_AAA_LORA, C), 0.1))
             self.a0 = nn.Parameter(torch.zeros(1,1,C)-0.19 + zigzag*0.3 + linear*0.4)
 
-            D_MV_LORA = max(32, int(round(  (1.7*(C**0.5))  /32)*32)) # suggestion
+            D_MV_LORA = getattr(args, "mv_lora_rank", max(32, int(round(  (1.7*(C**0.5))  /32)*32))) # suggestion
             self.v1 = nn.Parameter(torch.zeros(C, D_MV_LORA))
             self.v2 = nn.Parameter(ortho_init(torch.zeros(D_MV_LORA, C), 0.1))
             self.v0 = nn.Parameter(torch.zeros(1,1,C)+0.73 - linear*0.4)
 
             # Note: for some data, you can reduce D_GATE_LORA or even remove this gate
-            D_GATE_LORA = max(32, int(round(  (5*(C**0.5))  /32)*32)) # suggestion
+            D_GATE_LORA = getattr(args, "gate_lora_rank", max(32, int(round(  (5*(C**0.5))  /32)*32))) # suggestion
             self.g1 = nn.Parameter(torch.zeros(C, D_GATE_LORA))
             self.g2 = nn.Parameter(ortho_init(torch.zeros(D_GATE_LORA, C), 0.1))
